tests/unit/agent/adk/test_mcp_tool_loading.py

```diff
@@ -0,0 +1,57 @@
+"""Tests for MCP tool loading in setup.py with tool_name_prefix."""
+import pytest
+from unittest.mock import Mock, patch, AsyncMock
+from solace_agent_mesh.agent.adk.setup import _load_mcp_tool
+
+
+@pytest.fixture
+def mock_component():
+    """Mock SamAgentComponent for testing."""
+    component = Mock()
+    component.log_identifier = "[TestAgent]"
+    return component
+
+
+class TestMCPToolLoadingWithPrefix:
+    """Test _load_mcp_tool with tool_name_prefix parameter."""
+
+    @pytest.mark.asyncio
+    async def test_load_mcp_tool_with_prefix(self, mock_component):
```

Comment:
I don't love these tests - there is a pile of stuff just to say that, yes, the value passed into the function was actually passed in. Is it possible to actually test that setting the prefix actually causes the tool name to be changed?